### PR TITLE
Merge given branch with dev on compare request

### PR DESCRIPTION
**Pull request by wladekb**
As agreed we want to merge the tested branch with dev and run tests against such branch so the results will be closer to when you will eventually merge it.

/cc @harnash @jcellary

### CODE REVIEW

File: apps/compare_requests/views.py [hunk 1]
```diff
@@ -76,20 +77,16 @@ def create(self, request, *args, **kwargs):
         pull_info = github.get_pull_request_info(repo_name, pull_req_num)
```
Comment (jcellary) [inline apps/compare_requests/views.py]: This function has a total of 70 lines, that's quite lengthy and thus the code is not very readable. Wouldn't it make sense to split it into smaller abstractions? Your logger comments indicate the logical steps which I think would be a good starting point for division

File: apps/compare_requests/views.py [hunk 2]
```diff
@@ -99,23 +96,39 @@ def create(self, request, *args, **kwargs):
         else:
             logger.debug('  not found, creating new one...')
 
+            app_merged_branch_name = 'sparrow-test-{}-{}'.format(app_branch_info['sha'], app_dev_info['sha'])
+            logger.debug('Creating merged branch: {}'.format(app_merged_branch_name))
+            try:
+                app_merged_info = github.create_merged_ref(repo_name, app_merged_branch_name, app_branch_info['sha'],
+                                                              APP_TARGET_BRANCH)
+            except:
+                # assume the branch has been correctly created by another process
```
Comment (jcellary) [inline apps/compare_requests/views.py]: You assume but you do not check it in the exception - is that really the only case which can lead to generation of exception? What about IO errors?

**Comment by jcellary** [comment]
Code looks good I would just improve readability and exception handling as mentioned, other than that can be merged

File: apps/compare_requests/views.py (hunk 2 above)
Comment (wladekb) [inline apps/compare_requests/views.py]: Of course not and I thought to improve it later after we get a working end-to-end state.

File: apps/compare_requests/views.py [hunk 3]
```diff
@@ -71,56 +38,180 @@ def create(self, request, *args, **kwargs):
...
 
         github = GitHub()
+        create_compare_request = CreateCompareRequest(github, repo_name, pull_req_num)
+
+        try:
+            create_compare_request.execute()
+        except MergeFailed as e:
+            github.post_pull_request_comment(repo_name, pull_req_num, self.get_merge_failed_message())
+            raise
+
```
Comment (jcellary) [inline apps/compare_requests/views.py]: three empty lines in function body feel excessive, but maybe I'm being picky :)

**Comment by jcellary** [comment]
One more minor thing found, feel free to resolve either way and merge

**Comment by jcellary** [comment]
Now the code looks absolutely awesome!

But the unit tests are failing... :cry: Thank God we (I) did the github integration with jenkins for sparrow :laughing:

**Comment by wladekb** [comment]
Test failure is unrelated to the changes, it is a misconfiguration of the test environment.

**Pull request merged**